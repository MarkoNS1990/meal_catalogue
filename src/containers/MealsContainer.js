import React ,{useEffect} from 'react'
import {useSelector,useDispatch} from 'react-redux'
import { fetchMeals } from '../redux/meal/mealActions'
import Loader from "react-loader-spinner";
import Meal from './Meal';
import '../styles/MealsContainer.css'

function MealsContainer() {

    const dispatch = useDispatch()
    const meals = useSelector(state=>state.meals)
    const loading = useSelector(state=>state.loading)

    useEffect(()=>{
        dispatch(fetchMeals())
        
    },[dispatch])

    if(loading){
        return <Loader
        type="Puff"
        color="#rgb(194, 60, 60)"
        height={100}
        width={100}
      />
    }

    return (
        <div className='card-list'>
        {meals && meals.map(meal=><Meal key={meal.idMeal} meal={meal} />)}
            
        </div>
        
    )
}

export default MealsContainer
