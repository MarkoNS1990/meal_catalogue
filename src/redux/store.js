import {createStore,applyMiddleware} from 'redux';
import thunk from 'redux-thunk'
import mealReducer from './meal/mealReducer';

const store = createStore(mealReducer,applyMiddleware(thunk))

export default store