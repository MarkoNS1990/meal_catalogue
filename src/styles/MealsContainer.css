.card-list {
  display: flex;
  flex-wrap: wrap;
  margin: auto;
  justify-content: center;
}

.category {
  display: flex;
  flex-direction: column;
  align-items: center;
}

.form-select {
  width: 50%;
}
