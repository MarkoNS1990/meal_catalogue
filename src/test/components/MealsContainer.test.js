import React from 'react';
import { render, cleanup, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import {BrowserRouter} from 'react-router-dom'
import MealsContainer from '../../containers/MealsContainer';
import store from '../../redux/store';
import '@testing-library/jest-dom';

afterEach(cleanup);

it('should take a snapshot', () => {
  const { asFragment } = render(<BrowserRouter> <Provider store={store}><MealsContainer/></Provider></BrowserRouter>);
  expect(asFragment(<Provider store={store}><MealsContainer /></Provider>)).toMatchSnapshot();
});

it('displays correct first child', () => {
  const { container } = render(<BrowserRouter><Provider store={store}><MealsContainer /></Provider></BrowserRouter>);
  expect(container.firstChild).toMatchSnapshot();
});

it('negative check', () => {
  render(<BrowserRouter><Provider store={store}><MealsContainer /></Provider></BrowserRouter>);
  expect(screen.queryByText('some text!')).not.toBeInTheDocument();
});