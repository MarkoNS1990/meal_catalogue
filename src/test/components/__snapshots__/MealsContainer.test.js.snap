// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`displays correct first child 1`] = `
<div
  class="category mt-2"
>
  <h3
    class="category-h"
  >
    Select a category: 
  </h3>
  <select
    aria-label=".form-select-sm example"
    class="form-select form-select-sm"
    id="category"
    name="category"
  >
    <option
      value="Seafood"
    >
      Seafood
    </option>
    <option
      value="Dessert"
    >
      Dessert
    </option>
    <option
      value="Vegan"
    >
      Vegan
    </option>
    <option
      value="Breakfast"
    >
      Breakfast
    </option>
  </select>
</div>
`;

exports[`should take a snapshot 1`] = `
<DocumentFragment>
   
  <div
    class="category mt-2"
  >
    <h3
      class="category-h"
    >
      Select a category: 
    </h3>
    <select
      aria-label=".form-select-sm example"
      class="form-select form-select-sm"
      id="category"
      name="category"
    >
      <option
        value="Seafood"
      >
        Seafood
      </option>
      <option
        value="Dessert"
      >
        Dessert
      </option>
      <option
        value="Vegan"
      >
        Vegan
      </option>
      <option
        value="Breakfast"
      >
        Breakfast
      </option>
    </select>
  </div>
  <div
    class="card-list"
  />
</DocumentFragment>
`;
